internal/resource/factory.go

```diff
@@ -39,7 +39,30 @@ func WithConfig(manager Manager, config *spec.Config) Manager {
 }
 
 // getManager returns the resource manager depending on the system configuration.
-func getManager() Manager {
+func getManager(mode string) Manager {
+
+	resolved := resolveMode(mode)
+	switch resolved {
+	case "nvml":
+		klog.Info("Using NVML manager")
+		return NewNVMLManager()
+	case "tegra":
+		klog.Info("Using CUDA manager")
+		return NewCudaManager()
+	case "vfio":
+		klog.Info("Using Vfio manager")
+		return NewVfioManager()
+	}
+
+	klog.Warning("No valid resources detected; using empty manager.")
```

Comment:
nit: We may want to include `resolved` here:
```suggestion
	klog.Warningf("Unsupported mode detected: %v; using empty manager.", resolved)
```

Reply:
Fixed